### DISCUSSION TITLE
Give users words based on the letter(s) they select.

### DISCUSSION THREAD

**Discussion by zebbyjz** [monkeytypegame/monkeytype#765] [Ideas] [answered]
How about pick all letters from the language which contain the letters the user wants to practice. For example my average speed for "s" and "b" is very slow. So it only shows words which contain those letters. Might be a cool update.

**Comment by ryan-harrington** [comment]
This could also be based on key combinations that typists may struggle with whether it is their accuracy or speed.  I like the idea of having a "practice" mode that uses this idea. Additionally, practice mode could be infinite or for a set amount of time chosen by the user. As users type (or on reload), their progress is tracked and the practice words are updated.

**Comment by lucas-fern** [comment]
yeah this is a cool feature suggestion, could be a fair bit of work to implement in a meaningful way, though what you've suggested with taking words only if they contain a specific letter probably wouldn't be so hard. If you are looking specifically for this kind of feature I can recommend [KeyBr](www.keybr.com) and then coming back to MonkeyType once your accuracy is getting good, that's how I started improving at least.

**Comment by Miodec** [comment, marked as answer]
Interesting idea, might add it in the future. For now you can use https://gfruit.github.io/typing/words-filter.html to get a word list and use custom mode to practice them.

**Comment by ghost** [comment]
This feature is exactly what I'm looking for. This could work well in conjunction with stats which show the keys or key combinations that are quickest or slowest, potentially in a similar way to keybr.com. Presumably, this data powers the "weakspot" funbox, so it would be helpful to be able to see this data in a graph too.
That way you could then focus on those weakspots by picking those specific letters you want to practice like in the original idea post. That would be easier to use than trying to create an exhaustive word list filter anytime you want to practice 1 or more problem characters?

**Reply by Miodec** [reply]
Already added - you can use custom mode > words filter: 
![image](https://user-images.githubusercontent.com/13181393/166480553-007e8deb-1d55-488b-8bc1-556f71d664a0.png)
![image](https://user-images.githubusercontent.com/13181393/166480604-7b57bbaf-4f64-4931-9ac9-669d8b92190e.png)
